bazel: errcheck lies about the location of the problem

**Describe the problem**

We added this `nogo` thing recently to run some analysis tools. The errors are nice and it's nice that it's part of the build. Sadly, the locations are somehow totally wrong. This can be infuriating because you have a problem that prevents things from building but you can't find it. 

**To Reproduce**

Consider the below hypothetical change:
```
--- a/pkg/sql/sqlstats/persistedsqlstats/controller.go
+++ b/pkg/sql/sqlstats/persistedsqlstats/controller.go
@@ -83,5 +83,6 @@ func (s *Controller) ResetClusterSQLStats(ctx context.Context) error {
                return err
        }
 
-       return resetSysTableStats("system.transaction_statistics")
+       resetSysTableStats("system.transaction_statistics")
+       return nil
 }
```
If now I run `bazel test`, I'll see:
```
ERROR: /Users/ajwerner/src/github.com/cockroachdb/cockroach/pkg/sql/sqlstats/persistedsqlstats/BUILD.bazel:5:11: GoCompilePkg pkg/sql/sqlstats/persistedsqlstats/persistedsqlstats.a failed: (Exit 1): builder failed: error executing command bazel-out/host/bin/external/go_sdk/builder compilepkg -sdk external/go_sdk -installsuffix darwin_amd64 -tags bazel,gss,bazel,gss -src ... (remaining 97 argument(s) skipped)

Use --sandbox_debug to see verbose messages from the sandbox
compilepkg: nogo: errors found by nogo during build-time code analysis:
/var/folders/x3/98m68b9s2yv2lrdf8rxcrzgc0000gq/T/rules_go_work-77930801/github.com/cockroachdb/cockroach/pkg/sql/sqlstats/persistedsqlstats/flush.go:8:34: unchecked error (errcheck)
Target //pkg/sql:sql_test failed to build
Use --verbose_failures to see the command lines of failed build steps.
INFO: Elapsed time: 3.654s, Critical Path: 1.84s
INFO: 5 processes: 2 internal, 3 darwin-sandbox.
FAILED: Build did NOT complete successfully
//pkg/sql:sql_test                                              FAILED TO BUILD

FAILED: Build did NOT complete successfully
ERROR: exit status 1

```

`sql/sqlstats/persistedsqlstats/flush.go:8` isn't even the same file!

**Expected behavior**
It shows me the right location.

Epic CRDB-8036

--- a/build/patches/com_github_kisielk_errcheck.patch
+++ b/build/patches/com_github_kisielk_errcheck.patch
@@ -1,8 +1,11 @@
 diff -urN a/errcheck/analyzer.go b/errcheck/analyzer.go
 --- a/errcheck/analyzer.go
 +++ b/errcheck/analyzer.go
-@@ -6,6 +6,7 @@ import (
- 	"go/token"
+@@ -3,9 +3,9 @@ package errcheck
+ import (
+ 	"fmt"
+ 	"go/ast"
+-	"go/token"
  	"reflect"
  	"regexp"
 +	"strings"
@@ -41,3 +44,16 @@ diff -urN a/errcheck/analyzer.go b/errcheck/analyzer.go
  		excludes, err := ReadExcludes(argExcludeFile)
  		if err != nil {
  			return nil, fmt.Errorf("Could not read exclude file: %v\n", err)
+--- a/errcheck/analyzer.go
++++ b/errcheck/analyzer.go
+@@ -65,8 +65,9 @@ func runAnalyzer(pass *analysis.Pass) (interface{}, error) {
+ 		ast.Walk(v, f)
+ 
+ 		for _, err := range v.errors {
++			fsetFile := pass.Fset.File(f.Pos())
+ 			pass.Report(analysis.Diagnostic{
+-				Pos:     token.Pos(int(f.Pos()) + err.Pos.Offset),
++				Pos:     fsetFile.Pos(err.Pos.Offset),
+ 				Message: "unchecked error",
+ 			})
+ 		}

